model_compression_toolkit/exporter/model_exporter/tflite/int8_tflite_exporter.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2023 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import copy
+from typing import Callable
+
+import keras.models
+import numpy as np
+import tensorflow as tf
+from keras import Sequential
+from keras.layers import Dense, Conv2D, Reshape
+from keras.models import clone_model
+
+from model_compression_toolkit import quantizers_infrastructure as qi
+from model_compression_toolkit.core.common import Logger
+from model_compression_toolkit.exporter.model_exporter.keras.fakely_quant_keras_exporter import FakelyQuantKerasExporter
+
+BIAS_INITIALIZER = 'bias_initializer'
+BIAS_REGULARIZER = 'bias_regularizer'
+BIAS_CONSTRAINT = 'bias_constraint'
+ACTIVITY_REGULARIZER = 'activity_regularizer'
+KERNEL_INITIALIZER = 'kernel_initializer'
+KERNEL_REGULARIZER = 'kernel_regularizer'
+KERNEL_CONSTRAINT = 'kernel_constraint'
+KERNEL_SIZE = 'kernel_size'
+PADDING = 'padding'
+STRIDES = 'strides'
+LAYER_NAME = 'name'
+TRAINABLE = 'trainable'
+ACTIVATION = 'activation'
+USE_BIAS = 'use_bias'
+FILTERS = 'filters'
+UNITS = 'units'
+PAD_VALID = 'valid'
+
+
+class INT8TFLiteExporter(FakelyQuantKerasExporter):
+    """
+    Exporter for INT8 TFLite models.
+    The exporter expects to receive an exportable model (where each layer's full quantization parameters
+    can be retrieved), and convert it into a quantized model where weights and activations are represented
+    at integer data type.
+    """
+
+    def __init__(self,
+                 model: keras.models.Model,
+                 is_layer_exportable_fn: Callable,
+                 save_model_path: str):
+        """
+
+        Args:
+            model: Model to export.
+            is_layer_exportable_fn: Callable to check whether a layer can be exported or not.
+            save_model_path: Path to save the exported model.
+        """
+        super().__init__(model,
+                         is_layer_exportable_fn,
+                         save_model_path)
+
+        self.exported_model = None
+
+    def export(self) -> None:
+        """
+        Export a fully quantized model to its int8 tflite model.
+        """
+
+        def _substitute_model(wrapped_layer):
```

Comment:
Add description and type hints to this nested method?

Reply:
Also, maybe don't write it as a nested method, I don't see a reason for that and it makes it complicated to follow the exporter code